scheduler/server/persist_settings.go

```diff
@@ -0,0 +1,47 @@
+package server
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Persistor interface for persisting scheduler settings and initializing the scheduler
+// from its persisted settings
+type Persistor interface {
+	PersistSettings(s Scheduler)
```

Comment:
I'm confused by the interface and how it gets used by the scheduler. Assuming it makes sense that a Scheduler contains an implementation of Persistor, why would these methods take a Scheduler as a parameter? Instinctively I would expect to call something like:
`PersistSettings(ps PersistedSettings) error`
`LoadSettings() (ps PersistedSettings, error)`

Seems like at minimum these should return an error type as well - need to be able to propagate errors on set/get operations.

Reply:
agree that the interface was confusing... I reorganized it